### PR TITLE
Feature/annotation for header data

### PR DESCRIPTION
**Pull request by Jacu93**

### CODE REVIEW

File: src/main/java/com/piekoszek/nowaksharedrent/config/UndefinedExceptionHandlerController.java [hunk 1]
```diff
@@ -1,17 +1,12 @@
...
-import org.springframework.web.context.request.WebRequest;
 
 @ControllerAdvice
 class UndefinedExceptionHandlerController {
 
-    @ExceptionHandler(Throwable.class)
-    public ResponseEntity<Object> handleThrowableException (Throwable ex, WebRequest request) {
-        return new ResponseEntity<>(new MessageResponse("Undefined error! Sorry, we are working on that"), HttpStatus.BAD_REQUEST);
-    }
+//    @ExceptionHandler(Throwable.class)
```
Comment (piekaa) [inline src/main/java/com/piekoszek/nowaksharedrent/config/UndefinedExceptionHandlerController.java]: Why is it commented?

File: src/main/java/com/piekoszek/nowaksharedrent/jwt/JwtData.java [hunk 2]
```diff
@@ -11,6 +12,7 @@
 @Getter
 @Builder
 @EqualsAndHashCode
+@AllArgsConstructor
```
Comment (piekaa) [inline src/main/java/com/piekoszek/nowaksharedrent/jwt/JwtData.java]: @Builder is not enough?

Comment (Jacu93) [reply src/main/java/com/piekoszek/nowaksharedrent/jwt/JwtData.java]: I will check it today but as far as I remember there was error during building an app saying that default constructor is missing.

File: src/main/java/com/piekoszek/nowaksharedrent/config/UndefinedExceptionHandlerController.java (hunk 1 above)
Comment (Jacu93) [reply src/main/java/com/piekoszek/nowaksharedrent/config/UndefinedExceptionHandlerController.java]: I need some mechanism to log every undefined error. The problem is that I have no idea what's going on during debugging the app and I am commenting it everytime I'd like to check exception message. This time I forgot to uncomment it afterwards.

Comment (piekaa) [reply src/main/java/com/piekoszek/nowaksharedrent/config/UndefinedExceptionHandlerController.java]: You can log.error() this exception and you should see stack trace.

File: src/main/java/com/piekoszek/nowaksharedrent/jwt/JwtData.java (hunk 2 above)
Comment (Jacu93) [reply src/main/java/com/piekoszek/nowaksharedrent/jwt/JwtData.java]: Yup, builder was enough.

File: src/main/java/com/piekoszek/nowaksharedrent/config/UndefinedExceptionHandlerController.java (hunk 1 above)
Comment (Jacu93) [reply src/main/java/com/piekoszek/nowaksharedrent/config/UndefinedExceptionHandlerController.java]: Done

**Review by piekaa — APPROVED** [review]

**Pull request merged**